lib/core-classes/AssetCategoryService.ts

```diff
@@ -88,11 +89,27 @@ export class AssetCategoryService {
           throw global.app.errors.get('device-manager', 'asset_controller', 'mandatory_metadata', metadata.name);
         }
       }
-      if (metadata.valueList) {
-        if (! metadata.valueList.includes(assetMetadata[metadata.name])) {
-          throw global.app.errors.get('device-manager', 'asset_controller', 'enum_metadata', metadata.name, assetMetadata[metadata.name]);
+      this.validateEnumMetadata(metadata, assetMetadata[metadata.name]);
+    }
+  }
+
+  validateEnumMetadata (metadata, value) {
+    if (metadata.valueList) {
+      if (! metadata.valueList.includes(value)) {
+        throw global.app.errors.get('device-manager', 'asset_controller', 'enum_metadata', metadata.name, value);
+      }
+    }
+    if (metadata.objectValueList) {
+      let find = false;
+      for (const objectValue of metadata.objectValueList) {
+        const fromattedObjectValue = this.formatMetadataForGet(objectValue.object);
+        if (isEqual(fromattedObjectValue, value)) {
+          find = true;
```

Comment:
if find is true, it will never be false in the future, so you could break the for loop here